build.gradle

```diff
@@ -36,9 +46,95 @@ allprojects {
 
         mavenCentral()
     }
+
+    publishing {
+        publications {
+            mavenJava(MavenPublication) {
+                from components.java
+
+                artifactId project.name
+                groupId project.group
+                version project.version
+
+                pom {
+                    name = ''
+                    description = ''
+                    url = 'https://www.springwolf.dev'
+
+                    licenses {
+                        license {
+                            name = 'The Apache License, Version 2.0'
+                            url = 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+                        }
+                    }
+
+                    developers {
+                        developer {
+                            id = 'stavshamir'
+                            name = 'Stav Shamir'
+                            email = 'shamir.stav@gmail.com'
+                        }
+
+                        developer {
+                            id = 'timonback'
+                            name = 'Timon Back'
+                            email = 'timon.back+springwolf@gmail.com'
+                        }
+
+                        developer {
+                            id = 'sam0r040'
+                            name = 'David Müller'
+                            email = 'david.mueller@codecentric.de'
+                        }
+                    }
+
+                    scm {
+                        url = 'https://github.com/springwolf/springwolf-core'
+                    }
+                }
+            }
+        }
+
+        repositories {
+            maven {
+                def releasesRepoUrl = 'https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/'
+                def snapshotsRepoUrl = 'https://s01.oss.sonatype.org/content/repositories/snapshots/'
+                url = isSnapshot ? snapshotsRepoUrl : releasesRepoUrl
+                credentials {
+                    username = project.findProperty('MAVEN_USERNAME') ?: ''
+                    password = project.findProperty('MAVEN_PASSWORD') ?: ''
+                }
+            }
+        }
+    }
+
+    signing {
+        required { gradle.taskGraph.hasTask("publish") }
+        def signingKey = project.findProperty("SIGNINGKEY")
+        def signingPassword = project.findProperty("SIGNINGPASSWORD")
+        useInMemoryPgpKeys(signingKey, signingPassword)
+        sign publishing.publications.mavenJava
+    }
 }
 
 dependencyCheck {
     outputDirectory = "$buildDir/reports/owasp-dependency-check"
     formats = [ReportGenerator.Format.SARIF.toString()]
-}
\ No newline at end of file
+}
+
+def static getenv(path = ".env") {
```

Comment:
Can we rename this to some like `readEnvFile` or similar?
getenv seems to be related to a single env variable: https://man7.org/linux/man-pages/man3/getenv.3.html

For everyone not so familiar with gradle, it might be good to define a function for the line above as well:
`getenv().each { key, value -> if (System.getProperty(key as String) == null) { System.setProperty(key as String, value as String) } }`
Maybe `applyDotEnvFileAsGradleProperties()`?

Reply:
@timonback modified the code with the proposed approach